src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -115,6 +155,161 @@ public override async Task<WorkflowHandle<TWorkflow, TResult>> StartWorkflowAsyn
+                        input.Options.StartWorkflowOperation.Workflow,
+                        input.Options.StartWorkflowOperation.Args,
+                        input.Options.StartWorkflowOperation.Options,
+                        input.Options.StartWorkflowOperation.Headers).ConfigureAwait(false);
+                    var updateReq = await CreateUpdateWorkflowRequestAsync(
+                        input.Update,
+                        input.Args,
+                        input.Options,
+                        input.Options.WaitForStage,
+                        input.Headers).ConfigureAwait(false);
+                    updateReq.WorkflowExecution = new() { WorkflowId = startReq.WorkflowId };
+                    var req = new ExecuteMultiOperationRequest() { Namespace = Client.Options.Namespace };
+                    req.Operations.Add(
+                        new ExecuteMultiOperationRequest.Types.Operation() { StartWorkflow = startReq });
+                    req.Operations.Add(
+                        new ExecuteMultiOperationRequest.Types.Operation() { UpdateWorkflow = updateReq });
+
+                    // Continually try to start until an exception occurs, the user-asked stage is
+                    // reached, or the stage is accepted. But we will set the workflow handle as soon as
+                    // we can.
+                    UpdateWorkflowExecutionResponse? updateResp = null;
+                    string? runId = null;
+                    do
+                    {
+                        var resp = await Client.Connection.WorkflowService.ExecuteMultiOperationAsync(
+                            req, DefaultRetryOptions(input.Options.Rpc)).ConfigureAwait(false);
+                        // Set start result if not already set
+                        runId = resp.Responses[0].StartWorkflow.RunId;
+                        if (!input.Options.StartWorkflowOperation.IsCompleted)
+                        {
+                            input.Options.StartWorkflowOperation.SetResult((WorkflowHandle)Activator.CreateInstance(
+                                input.Options.StartWorkflowOperation.HandleType,
+                                // Parameters on workflow handles are always: client, ID, run ID,
+                                // result run ID, and first execution run ID. This code is tested to
+                                // confirm.
+                                Client,
+                                input.Options.StartWorkflowOperation.Options.Id!,
+                                null,
+                                runId,
+                                runId)!);
+                        }
+                        if (resp.Responses.Count > 1)
+                        {
+                            updateResp = resp.Responses[1].UpdateWorkflow;
+                        }
+                    }
+                    while (updateResp == null || (
```

Comment:
```suggestion
                        updateResp = resp.Responses[1].UpdateWorkflow;
                    }
                    while (
```

Looking at the Python PR, this response is always present even if the update is not yet done

Reply:
Fixed